### PR TITLE
Dockerise sbt

### PR DESCRIPTION
**Pull request by wjlow**

### CODE REVIEW

File: auto/compile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -e
+
+cd $(dirname $0)/..
+
+auto/sbt sbt compile
```
Comment (sanjivsahayamrea) [inline auto/compile]: do we need to call sbt twice here?

File: support/Dockerfile [hunk 2]
```diff
@@ -0,0 +1 @@
+FROM hseeberger/scala-sbt:8u181_2.12.6_1.2.1
```
Comment (shaun-whitely) [inline support/Dockerfile]: If you're not adding any additional things to the `scala-sbt` image, I'd suggest just saying `image: hseeberger/scala-sbt:8u181_2.12.6_1.2.1` in your `docker-compose`. Also, have a look at [dfresh](https://github.com/realestate-com-au/dfresh) to lock down the exact image so everyone is guaranteed to have the same environment.

File: auto/sbt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+# Create volume
+docker volume create --name ivy-cache > /dev/null
+docker volume create --name docker-scala-sbt-cache > /dev/null
+
+# rebuild if necessary
+docker-compose build dev
+
+# Run dev script or use what is passed in as arguments
+docker-compose run --rm --service-ports dev "$@"
```
Comment (shaun-whitely) [inline auto/sbt]: A common pattern at REA is to name a script like this `auto/dev-environment`. Then you can make the `docker-compose` command `docker-compose run --rm --service-ports dev "${@:-bash}" so it'll run bash by default. Then you can create an `auto/sbt` script which does something like `auto/dev-environment sbt $@`. Then you could do as @sanjivsahayamrea suggested.

Comment (sanjivsahayamrea) [reply auto/sbt]: Good idea @shaun-whitely. We do call it `dev-environment` in Listings.

**Comment by wjlow** [comment]
I've decided to take the risk of locking down to a tag :)

**Pull request merged**